### PR TITLE
Adds handler for Rails 5.1

### PR DESCRIPTION
**Pull request by sjke**

### CODE REVIEW

File: lib/pg_ltree.rb [hunk 1]
```diff
@@ -2,4 +2,14 @@
...
 
-ActiveRecord::Base.extend(PgLtree::Ltree)
+Dir[File.expand_path('../pg_ltree/versions/**/*.rb', __FILE__)].each { |file| require file }
+
+if defined?(ActiveRecord)
+  ActiveRecord::Base.extend(PgLtree::Ltree)
+
+  # The behavior of _was changes in Rails 5.1
+  # http://blog.toshima.ru/2017/04/06/saved-change-to-attribute.html
+  if ActiveRecord::VERSION::MAJOR > 5 || (ActiveRecord::VERSION::MAJOR == 5 && ActiveRecord::VERSION::MINOR >= 1)
```
Comment (HoyaBoya) [inline lib/pg_ltree.rb]: Nice!

File: lib/pg_ltree/versions/rails_5_1.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        # https://github.com/rails/rails/pull/25337
+        #
+        # @return [String] ltree previous value
+        def ltree_path_was
+          public_send :"#{ltree_path_column}_before_last_save"
+        end
+
+        #
+        # In order for for cascade_destroy to work with the current callbacks, let's first delete the column :/.
+        # @author HoyaBoya [https://github.com/HoyaBoya]
```
Comment (HoyaBoya) [inline lib/pg_ltree/versions/rails_5_1.rb]: Thanks for the attribution but not needed if you want to keep the comments clean.

**Comment by sb1752** [comment]
👏

**Comment by HoyaBoya** [comment]
👍

**Pull request merged**